### PR TITLE
fix(span-metrics): Scrub subdomains in metric tags

### PR DESCRIPTION
**Pull request by iker-barriocanal**
A dummy and straight-forward approach to scrub subdomains and reduce metric cardinality. It's a best effort.

#skip-changelog

### CODE REVIEW

File: relay-server/src/metrics_extraction/transactions/mod.rs [hunk 1]
```diff
@@ -479,7 +479,23 @@ fn extract_span_metrics(
...
+                                            tokens.iter().rev().take(2).rev().collect();
+                                        Some(format!(
+                                            "*.{}.{}:{}",
+                                            extracted_domain[0], extracted_domain[1], port
+                                        ))
+                                    } else {
+                                        Some(format!("{}:{}", domain, port))
+                                    }
+                                }
                                 (Some(domain), None) => Some(domain),
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/transactions/mod.rs]: Shouldn't we handle the case when there's no port?

File: relay-server/src/metrics_extraction/transactions/mod.rs [hunk 2]
```diff
@@ -479,7 +479,23 @@ fn extract_span_metrics(
...
+                                        // <name>.service.<cloud-region>.<whatever>
+                                        let extracted_domain: Vec<&&str> =
+                                            tokens.iter().rev().take(2).rev().collect();
+                                        Some(format!(
+                                            "*.{}.{}:{}",
+                                            extracted_domain[0], extracted_domain[1], port
+                                        ))
+                                    } else {
+                                        Some(format!("{}:{}", domain, port))
+                                    }
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/transactions/mod.rs]: nit: Should be possible to do this without allocating a vector (and without the `else` branch), but I don't have the code for it ready so feel free to ignore.

**Review by jjbayer — CHANGES_REQUESTED** [review]
Requesting changes because of the no port case

File: relay-server/src/metrics_extraction/transactions/mod.rs [hunk 3]
```diff
@@ -543,6 +542,33 @@ fn extract_span_metrics(
...
+    // this and consider cloud domains, e.g.
+    // <name>.service.<cloud-region>.<whatever>
+    let extracted_domain: Vec<&&str> = tokens.iter().rev().take(2).rev().collect();
+    if let Some(p) = port {
+        return Some(format!(
+            "*.{}.{}:{}",
+            extracted_domain[0], extracted_domain[1], p
+        ));
+    }
+    return Some(format!("*.{}.{}", extracted_domain[0], extracted_domain[1]));
```
Comment (iker-barriocanal) [inline relay-server/src/metrics_extraction/transactions/mod.rs]: It'd be more readable to generate more than one string, but I think the code isn't that complicated to make it worth the performance overhead of creating more strings.

File: relay-server/src/metrics_extraction/transactions/mod.rs [hunk 4]
```diff
@@ -543,6 +542,33 @@ fn extract_span_metrics(
...
+    }
+
+    // sth.subdomain.domain.tld => *.domain.tld
+    // TODO(iker): find a proper way to do
+    // this and consider cloud domains, e.g.
+    // <name>.service.<cloud-region>.<whatever>
+    let extracted_domain: Vec<&&str> = tokens.iter().rev().take(2).rev().collect();
+    if let Some(p) = port {
+        return Some(format!(
+            "*.{}.{}:{}",
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/transactions/mod.rs]: This adds `*.` even if the domain looks like `domain.com`.

File: relay-server/src/metrics_extraction/transactions/mod.rs [hunk 5]
```diff
@@ -543,6 +542,33 @@ fn extract_span_metrics(
...
+            extracted_domain[0], extracted_domain[1], p
+        ));
+    }
+    Some(format!("*.{}.{}", extracted_domain[0], extracted_domain[1]))
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/transactions/mod.rs]: What do you think of this simplification?
```suggestion
    let mut parts = domain.rsplitn(3, '.');
    let tld = parts.next();
    let domain = parts.next();
    let prefix = parts.next().map(|_| "*");

    let mut replaced = prefix
        .iter()
        .chain(domain.iter())
        .chain(tld.iter())
        .join(".");
    if let Some(port) = port {
        replaced = format!("{}:{}", replaced, port);
    }
    Some(replaced)
```

**Review by jjbayer — CHANGES_REQUESTED** [review]

**Review by jjbayer — APPROVED** [review]

**Pull request merged**